### ISSUE TITLE
Setup script does not stage sfcanal_data.tile warm-start ICs

### ISSUE THREAD

**Issue by TravisElless-NOAA** [NOAA-EMC/global-workflow#2742]
### What is wrong?

When using `--icsdir` to stage warm-start initial conditions as part of `setup_expt.py`, the `sfcanl_data.tile` files located in the `CDATE` `model_data/atmos/restart/` directory are not being staged in the comroot. 

### What should have happened?

All files needed to warm-start an experiment should be gathered by `setup_expt.py`

### What machines are impacted?

All or N/A

### Steps to reproduce

1. Run `setup_expt.py` for a warm-start experiment using the `--icsdir` option.
2. Check what files are present in the comroot. 

### Additional information

Example output for a 3DVAR test from running this setup script is located at `/scratch2/NCEPDEV/stmp3/Travis.J.Elless/comrot/test_setup` using ICs from `/scratch2/NCEPDEV/stmp3/Travis.J.Elless/comrot/test_warm_start/` and `--idate 2021080500`

**Comment by aerorahul** [comment]
`sfc_data.tileX.nc` is the warmstart file.
`sfcanl_data.tileX.nc` is the analysis tile file (not a warm start).
Your best bet is to stage the `sfcanl_data.tileX.nc` replacing the `sfc_data.tileX.nc` and then execute `setup_expt.py`.

**Comment by TravisElless-NOAA** [comment]
Given that these files are included in the restarta files and `setup_expt.py` grabs other analysis files from this restart group, shouldn't it be grabbing the analysis surface files as well? Without these files, we lose reproducibility when restarting simulations.

**Comment by aerorahul** [comment]
Yes. We should.  The situation you are describing is an identical start from a previous experiment.
We did not have a requirement such as that.  We can add a feature request for it in the future.

**Comment by aerorahul** [comment]
We are working on a refactoring of the `setup_expt.py` that will drive the files one needs from a yaml.  
It will then be up to the user to describe the files they want to copy from/to.

**Comment by WalterKolczynski-NOAA** [comment]
@KateFriedman-NOAA please confirm this has been solved.

**Comment by KateFriedman-NOAA** [comment]
This has been resolved. The updated staging job now includes the `sfcanl_data.tile` files if they exist in the `ICSDIR`:

```
{% if path_exists(ICSDIR ~ "/" ~ COMOUT_ATMOS_RESTART_PREV_MEM | relpath(ROTDIR) ~ "/" ~ p_prefix ~ ".sfcanl_data.tile" ~ ntile ~ ".nc") %}
- ["{{ ICSDIR }}/{{ COMOUT_ATMOS_RESTART_PREV_MEM | relpath(ROTDIR) }}/{{ p_prefix }}.sfcanl_data.tile{{ ntile }}.nc", "{{ COMOUT_ATMOS_RESTART_PREV_MEM }}"]
{% endif %} # path_exists
```
https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/stage/atmosphere_warm.yaml.j2#L24:L26

Closing this as complete.

**Comment by CatherineThomas-NOAA** [comment]
Reopening issue.

@XuanliLi-NOAA found that the stage ic job was not able to reproduce a previous experiment when warm starting.  Upon closer inspection, we found that the `sfcanl_data` tiles were the culprit.  The `atmosphere_warm.yaml.j2` file contains the following:
```
     	{% if path_exists(ICSDIR ~ "/" ~ COMOUT_ATMOS_RESTART_PREV_MEM | relpath(ROTDIR) ~ "/" ~ p_prefix ~ ".sfcanl_data.tile" ~ ntile ~ ".nc") %}
     	- ["{{ ICSDIR }}/{{ COMOUT_ATMOS_RESTART_PREV_MEM | relpath(ROTDIR) }}/{{ p_prefix }}.sfcanl_data.tile{{ ntile }}.nc", "{{ COMOUT_ATMOS_RESTART_PREV_MEM }}"]
```

This is looking in the previous cycle’s restart directory for the `sfcanl_data` tiles, when they are actually located in the current cycle’s restart directory:
```
     	{% if path_exists(ICSDIR ~ "/" ~ COMOUT_ATMOS_RESTART_MEM | relpath(ROTDIR) ~ "/" ~ m_prefix ~ ".sfcanl_data.tile" ~ ntile ~ ".nc") %}
     	- ["{{ ICSDIR }}/{{ COMOUT_ATMOS_RESTART_MEM | relpath(ROTDIR) }}/{{ m_prefix }}.sfcanl_data.tile{{ ntile }}.nc", "{{ COMOUT_ATMOS_RESTART_MEM }}"]
```
Xuanli tested this change (along with supporting variable definitions) with her C384/C192 ATM-only case on Hera and was able to reproduce a previously running experiment.  I’m currently testing the new branch [feature/stage](https://github.com/CatherineThomas-NOAA/global-workflow/tree/feature/stage) with C96/C48 on WCOSS.  I’ll open the PR once I confirm it works as intended.

**Comment by CatherineThomas-NOAA** [comment]
I've confirmed that a 2.5-cycle coldstart C96/C48 ATM-only case can be reproduced by using this branch and it's stage_ic job to warm start from the first full cycle.
